src/sdg/producten/tests/test_update_catalogs.py

```diff
@@ -1,27 +1,327 @@
+            organisatie__owms_end_date=None,
+        )
+
+        self.assertEqual(producten_table.count(), 1)
+
+        call_command("update_catalogs")
+
+        new_product = producten_table.get(referentie_product=ref_product)
+        new_product_versie = producten_versies_table.get(product=new_product)
+        new_localized_product_nl = localized_producten_table.get(
+            product_versie=new_product_versie, taal="nl"
+        )
+
+        self.assertEqual(producten_table.count(), 2)
+        self.assertEqual(new_product.catalogus.lokale_overheid, lokale_overheid)
+        self.assertEqual(
+            ref_localized_product_nl.specifieke_tekst,
+            new_localized_product_nl.specifieke_tekst,
+        )
 
     def test_no_create_products_from_reference_without_active_version(self):
+        # if product has no publicatie date check if we don't generate any products ?
         pass
 
     def test_correct_products_if_no_initial_version(self):
-        pass
+        producten_table = Product.objects.all()
+
+        ref_overheid = LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=False
+        )
+        ref_catalog = ProductenCatalogusFactory.create(
+            lokale_overheid=ref_overheid,
+            is_referentie_catalogus=True,
+        )
+        ref_product = ReferentieProductFactory.create(catalogus=ref_catalog)
+
+        lokale_overheid = LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=True,
+            organisatie__owms_end_date=None,
+        )
+
+        self.assertEqual(producten_table.count(), 1)
+        call_command("update_catalogs")
+        new_product = producten_table.get(referentie_product=ref_product)
+
+        self.assertEqual(producten_table.count(), 2)
+        self.assertEqual(new_product.catalogus.lokale_overheid, lokale_overheid)
 
     def test_correct_products_if_no_translations(self):
-        pass
+        localized_producten_table = LocalizedProduct.objects.all()
+
+        ref_overheid = LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=False
+        )
+        ref_catalog = ProductenCatalogusFactory.create(
+            lokale_overheid=ref_overheid,
+            is_referentie_catalogus=True,
+        )
+        ref_product = ReferentieProductFactory.create(
+            catalogus=ref_catalog,
+        )
+        ProductVersieFactory.create(
+            product=ref_product,
+            publicatie_datum=date.today(),
+        )
+
+        LokaleOverheidFactory.create(
+            automatisch_catalogus_aanmaken=True,
+            organisatie__owms_end_date=None,
+        )
+
+        self.assertEqual(localized_producten_table.count(), 0)
+        call_command("update_catalogs")
+        self.assertEqual(localized_producten_table.count(), 2)
+        self.assertEqual(localized_producten_table.first().taal, "nl")
+        self.assertEqual(localized_producten_table.last().taal, "en")
```

Comment:
Don't rely on ordering!!

Pseudo-code!

```suggestion
        self.assertSetEqual(set(LocalizedProduct.objects.filter(catalogus__lokale_overheid=my_lokale_overheid).values_list("taal", flat=True)), {"nl", "en"})
```

Note that I again don't use that localized_producten_table. Most of all, because I need to scroll up to see what it did, and second, its the entire table while we need to check for localized products on a specific catalog/lokale overheid.